### DISCUSSION TITLE
Feedback round for 0.6.0 - hobbyist

### DISCUSSION THREAD

**Discussion by matmair** [inventree/InvenTree#2375] [Ideas]
Hi there @sintech!

You had a lot of ideas that you mentioned in #1471. A fair amount of them was implemented since April so I wanted to ask your opinion on the upcoming 0.6.0 release.
Would you be willing to check the current master and provide some feedback? I think your input was valuable and valued in the past and I would like to get the perspective of a hobbyist user.

Thanks Matthias

**Comment by sintech** [comment]
First of all thank you @SchrodingersGat @matmair @eeintech and all other developers for keep on working on Inventree.
I know a lot was changed under the hood but let me share what I noticed as a user.

Improvements:
- #2074, looks very promising, can't wait to see it on my instance. I have a couple of scripts for new part info import that I run standalone and communicate to Inventree via API. Good candidates for plugin.
- #2373 is a step in the right direction but multiple images for Part is more important IMHO.
- #2085 - nice customization option.
- #2150 is a feature I really missed. And it looks almost exactly like I imagined.

Don't know why, but interesting (for me) PRs got stalled:
- #2196, purchase price history. Currently price info disappears with the last stock item. 
- #1730, "additional line items" is a really good idea.
	
Corner of criticism:
- Personally I don't like new look of inventree since Bootstrap 5 was introduced. Maybe it is a new color scheme, or bulky buttons and other UI elements but It just doesn't feel as good as before.
__It would be nice to have "Inventree classic" theme (if it is possible).__
- Missing category/stock location sidebar, already discussed, clearly a regression.  #2250
- New Part screen looks strange, too many empty space with UI controls scattered on the opposite sides of the screen. What If I have 27" monitor, my neck would hurt a lot. ;) <details>
![изображение](https://user-images.githubusercontent.com/10057327/143849004-047454e6-23dc-42d5-861e-4969ac617f11.png)
</details>

Regards,
Konstantin

**Reply by eeintech** [reply]
I have to agree that the new Part page is definitely not my favorite, I actually like the ability to hide/show part details on all tabs, and the parameters being the default view (eg. `0.5.4` view is much better IMO). After seeing this, I feel it is one of the reason I'm going to keep `0.5.x` around for a while at our company as this kind of UI disruption could confuse some, on top of the fact that I don't like it...

**Reply by matmair** [reply]
> Don't know why, but interesting (for me) PRs got stalled:

@sintech Well, I do not really have a use for the pricing features myself and I found other stuff more interesting. Intrinsic motivators help a lot with the development effort.
I am really more focused on security and the plugin ecosystem. I think that will bring much more value to the whole system. Especially if ppl can develop without diving deep into the system



> * It would be nice to have "Inventree classic" theme (if it is possible).

There is theming support, have you tried out the other themes? It is pretty easy to add a new theme, it is just one CSS (maybe look at the docs about themeing)



> After seeing this, I feel it is one of the reason I'm going to keep `0.5.x` around for a while at our company

@eeintech what are the other big reasons?

**Reply by eeintech** [reply]
@matmair People at my company are already used to the current interface (eg. `0.5.x`) and the BS5 update has changed it a lot. I don't feel comfortable updating to master now because of it might create a lot of confusion and I have not a lot of time to spend on our server maintenance anymore. I usually prioritize stable software so once I see that the dust settles on the UI and that `0.6.x` and later do not move things around then I might kick-off the migration to the new UI.

**Reply by SchrodingersGat** [reply]
@eeintech it is a large set of visual changes, but it will definitely stabilize for the 0.6.x release

**Reply by matmair** [reply]
@eeintech going back to 0.5.x paradigms does not really make sense, there is a lot of technical debt there. The old interface is full of static server-side rendering and a very old bootstrap that does not support a lot of modern ways to work incl. responsiveness (we already run on 0.6.x in some labs and it is great with old iPads for example).
Some input on what should be changed so the switch is easier would be greatly appreciated.

**Reply by eeintech** [reply]
@matmair I do get what you are saying but even though it is _technically_ better, it does not make it inherently more user friendly and less disruptive to toggle the switch, people with less technical abilities might not adapt seamlessly to the switch like you and have would. If the interface was closer to what's on `0.5.4` for the part page, that would make me more comfortable switching. I was really pleased and got attached to the layout on that page and sad to see it changed so much:

## master

![image](https://user-images.githubusercontent.com/4020546/143947778-526fdea9-c5ff-4981-8010-6f4fbcb54063.png)

## 0.5.4

![image](https://user-images.githubusercontent.com/4020546/143947740-7428e4de-9883-41ca-b3c2-12ec539fdc9e.png)

**Reply by matmair** [reply]
@eeintech as far as I see it all actions are now always in the same place and the part details have their own tab. Does not seem to be that big of a change. We could bring back an option to show the part details on each page if that suits your needs.

**Reply by matmair** [reply]
And the star is now a bell as the function has changed to a more complex notification system that supports mail for now and probably web notifications when we get 0.6.x ready.

**Reply by eeintech** [reply]
Sure I can contribute to improving this page if I find time, having in the header row all the way to the right feels misplaced, I like when all the controls are closer to the zone of action which is the left hand side.

**Reply by matmair** [reply]
I think we should have a general UI/UX design discussion as @schrodingersGat and I just refactored all components (including coming up features) to follow that new paradigm.
One of the big reasons why I prefer it this way: it reduces the space that is needed overall while enlarging the touch targets to something usable. I maintain a few instances that run in mid-scale production and rapid-prototyping environments where a lot of mounted tablets and industrial touch-displays (think cnc-controller like) are used. InvenTree was unusable till the switch to bootstrap 5 (which is almost mobile first) and the refactor. There are a lot more formfactors than a 1080/1440 24/27 inch landscape-oriented screen with mouse input in manufacturing and I think we should consider that. Mice are just inefficient way to interact when you quickly need something.

**Comment by SchrodingersGat** [comment]
@sintech 

> Attachment links #2373 is a step in the right direction but multiple images for Part is more important IMHO.

Feel free to submit a PR with this functionality if you like :)

> Don't know why, but interesting (for me) PRs got stalled:

Limited time, and lots of open PRs! Planning to get to those, but *when* is the question...

> Missing category/stock location sidebar, already discussed, clearly a regression.

This will be added back in for 0.6.0 - the "old" way it was implemented was quite bad, so we need a "new" way ;)

> New Part screen looks strange, too many empty space with UI controls scattered on the opposite sides of the screen. What If I have 27" monitor, my neck would hurt a lot. ;)

I agree in part. The new visual refactor moved a lot of the action buttons into the "title" blocks of various screens - saving a lot of vertical space. However this has left a visual "negative space" where they used to be.. I think that there are some definite improvements that can be made here before the next release. If you have any particular suggestions please let me know

**Comment by SchrodingersGat** [comment]
@eeintech it's the price of progress ;)

**Reply by eeintech** [reply]
Indeed... but what is?
